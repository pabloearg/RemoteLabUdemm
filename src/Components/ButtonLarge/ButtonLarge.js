// @flow
import React from 'react';
import { StyleSheet } from 'react-native';
import type { ViewStyle, TextStyle } from '../../styles/types';
import Button from '../Button';

type Props = {
  backgroundColor?: string,
  style?: ViewStyle,
  onPress: ()=>any,
  children: any,
  buttonTextStyle: TextStyle,
  text?: string,
}

const ButtonLarge = (props: Props) => {
  const {
    backgroundColor,
    style,
    onPress,
    children,
    buttonTextStyle,
    text,
    ...other
  } = props;

  ButtonLarge.defaultProps = {
    backgroundColor: undefined,
    style: undefined,
    text: undefined,
  };

  return (
    <Button
      onPress={onPress}
      backgroundColor={backgroundColor}
      style={[styles.Button, props.style]}
      textStyle={buttonTextStyle}
      {...other}
    >
      {text ?? props.children}
    </Button>
  );
};

const styles = StyleSheet.create({
  Button: {
    padding: 16
  }
});

export default ButtonLarge;
