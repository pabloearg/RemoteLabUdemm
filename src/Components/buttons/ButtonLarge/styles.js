import { StyleSheet } from 'react-native';
import {
  BLACK,
  ORANGE,
  DISABLED_GREY,
  BLUE_HEADER,
} from '../../../styles/colors';

export const styles = StyleSheet.create({
  Button: {
    height: 50,
    shadowColor: BLACK,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.15,
    shadowRadius: 2,
    elevation: 1,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 10,
  },
  buttonTextStyle: {
    fontWeight: '900',
    fontSize: 14
  },
  disabled: {
    backgroundColor: DISABLED_GREY,

  },
  normal: {
    backgroundColor: BLUE_HEADER,
  }
});
