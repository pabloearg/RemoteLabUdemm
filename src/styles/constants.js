export const LOW_SIZE_PERCENTAGE = 7.46 / 100;
export const MEDIUM_SIZE_PERCENTAGE = LOW_SIZE_PERCENTAGE * 2;
export const HIGH_SIZE_PERCENTAGE = LOW_SIZE_PERCENTAGE * 3;
