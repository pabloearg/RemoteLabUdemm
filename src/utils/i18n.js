// @flow
import I18n from 'i18n-js';

import es from '../static/locale/locale-es';

I18n.defaultLocale = 'es';
I18n.locale = 'es';

I18n.translations = {
  es,
};

export default I18n;
