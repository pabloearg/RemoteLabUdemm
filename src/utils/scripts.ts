import moment from "moment"
import { v4 as uuidv4 } from 'uuid';
export const getAppointmentsArray = (experimentId: string) => {

  const appointments = []
  for (let i = 0; i < 48; i++) {
    for (let hour = 9; hour < 15; hour++) {
      for (let minutes = 0; minutes < 60; minutes += 30) {
        const date = moment().set("hours", 0).set("minutes", 0)
        date.add(i, "days")
        date.add(hour, "hours")
        date.add(minutes, "minutes")
        const id = uuidv4()
        const appointment = {
          uuid: id,
          day: date.format("MM-DD-YYYY"),
          hour: date.format("hhmm"),
          isTaken: "0",
          experimentId: experimentId,
          id: id,
        }
        appointments.push(appointment)
      }

    }
  }
  return appointments
  console.log("appointments: ", appointments)
}