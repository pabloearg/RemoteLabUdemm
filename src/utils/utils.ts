import moment from 'moment';
import { Appointment } from '../API';

moment.locale('es-mx');

export const waitMillis = (timeInMillis: number): Promise<any> => new Promise((resolve) => setTimeout(resolve, timeInMillis));

export const convertExperimentArray = (array: any[]) => {
  const newObj = {};
  array.forEach((obj) => {
    newObj[obj.uuid] = obj;
  });
  return newObj;
};

export const getHourFromAppointment = (appointment: Appointment) => `${appointment?.hour?.substr(0, 2)}:${appointment?.hour?.substr(2, 2)}`;

export const getFormatedDayFromAppointment = (appointment: Appointment) => {
  if (!appointment?.day) {
    return '';
  }
  const momentDay = moment(appointment.day);
  return momentDay.format('DD MMM yyyy');
};
