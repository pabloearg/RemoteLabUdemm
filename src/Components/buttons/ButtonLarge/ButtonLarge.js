// @flow
import React from 'react';
import { StyleSheet } from 'react-native';
import type {
  ViewStyle,
  TextStyle,
} from '../../../styles/types';
import Button from '../Button';
import { styles } from './styles';

type Props = {
  style?: ViewStyle,
  buttonTextStyle?: TextStyle,
  onPress: ()=>any,
  text?: string,
  btnAccessibilityLabel: string,
  disabled: boolean,
}

const ButtonLarge = (props: Props) => {
  const {
    backgroundColor,
    style,
    buttonTextStyle,
    disabled,
    onPress,
    accessibilityLabel,
    text,
    ...other
  } = props;

  return (
    <Button
      onPress={onPress}
      backgroundColor={disabled ? styles.disabled.backgroundColor : styles.normal.backgroundColor}
      accessibilityLabel={accessibilityLabel}
      disabled={disabled}
      style={[styles.Button, style]}
      textStyle={[styles.buttonTextStyle, buttonTextStyle]}
      text={text}
      // eslint-disable-next-line react/jsx-props-no-spreading
      {...other}
    />
  );
};

ButtonLarge.defaultProps = {
  style: undefined,
  buttonTextStyle: undefined,
  text: undefined,
};

export default ButtonLarge;
