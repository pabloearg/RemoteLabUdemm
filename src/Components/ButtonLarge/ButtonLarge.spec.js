import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ButtonLarge from './ButtonLarge';
import TextStrong from '../TextStrong';
import { ORANGE } from '../../utils/constants';

describe('ButtonLarge', () => {
  describe('Rendering', () => {
    it('should match to snapshot - Primary', () => {
      const component = shallow(
        <ButtonLarge onPress={() => {}} backgroundColor={ORANGE}>Continue</ButtonLarge>
      );
      expect(toJson(component)).toMatchSnapshot('Primary button snapshot');
    });
    it('should match to snapshot - Secondary', () => {
      const component = shallow(
        <ButtonLarge onPress={() => {}}><TextStrong>Probando</TextStrong></ButtonLarge>
      );
      expect(toJson(component)).toMatchSnapshot('Secondary button snapshot');
    });
  });

  describe('Interaction', () => {
    describe('onPressHandler', () => {
      it('should call onPress', () => {
        const mockOnPress = jest.fn();
        const component = shallow(
          <ButtonLarge onPress={mockOnPress}>Probando</ButtonLarge>
        );
        component.props().onPress();

        expect(mockOnPress).toHaveBeenCalled();
        expect(mockOnPress).toHaveBeenCalledTimes(1);
      });
    });
  });
});
