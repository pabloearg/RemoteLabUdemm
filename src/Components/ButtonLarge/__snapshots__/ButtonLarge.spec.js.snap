// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLarge Rendering should match to snapshot - Primary: Primary button snapshot 1`] = `
<Button
  backgroundColor="#F3A802"
  onPress={[Function]}
  style={
    Array [
      Object {
        "padding": 16,
      },
      undefined,
    ]
  }
>
  Continue
</Button>
`;

exports[`ButtonLarge Rendering should match to snapshot - Secondary: Secondary button snapshot 1`] = `
<Button
  onPress={[Function]}
  style={
    Array [
      Object {
        "padding": 16,
      },
      undefined,
    ]
  }
>
  <TextStrong
    style={Object {}}
  >
    Probando
  </TextStrong>
</Button>
`;
